### PR TITLE
Compatibility with `snakemake>=8`

### PR DESCRIPTION
**Pull request by fneum**
https://snakemake.readthedocs.io/en/latest/getting_started/migration.html

- [x] adjustment of `_helpers.mock_snakemake()` function to new API
- [x] use of new storage plugins for remote files (https://snakemake.readthedocs.io/en/latest/snakefiles/storage.html#storage-support)
- [x] verify cluster integration
- [x] do a PyPSA-Eur release before merging as this bumps the minimum snakemake version to v8

### CODE REVIEW

**Comment by FabianHofmann** [comment]
I think we can soon merge this right? (after solving the merge conflicts)

**Comment by fneum** [comment]
yes, I just need to check if the cluster setup still works or what has to be adapted to merge.

**Review by FabianHofmann — APPROVED** [review]
awesome!

**Pull request merged**
